packages/@atjson/document/src/attributes.ts

```diff
@@ -100,3 +100,15 @@ export function clone(attribute: any): NonNullable<any> {
     return attribute;
   }
 }
+
+export function removeUndefinedValuesFromObject(obj: any): any {
+  const newObj = { ...obj };
+  Object.keys(newObj).forEach(key => {
+    if (newObj[key] && typeof newObj[key] === "object") {
+      removeUndefinedValuesFromObject(newObj[key]);
+    } else if (newObj[key] === undefined) {
+      delete newObj[key];
```

Comment:
`delete` changes the shape of the object in V8 and we prefer _not_ to use this.

Reply:
hmm not sure I understand. Wouldn’t changing the shape of the object be the goal if we're removing the undefined values from it?

Reply:
Oh, by shape I mean the shape that V8 sees. It's not _super_ important here, but for consistency, it's probably best not to use `delete`.

Reply:
Abstractly, "shape" = V8 optimized "struct"

Reply:
@tim-evans updated the logic to remove the `delete` here.